src/common/Gateway.sol

```diff
@@ -448,6 +448,9 @@ contract Gateway is Auth, Recoverable, IGateway {
     /// @inheritdoc IGateway
     function endBatching() external auth {
         require(isBatching, NoBatched());
+        
+        TransientArrayLib.clear(BATCH_LOCATORS_SLOT);
+        isBatching = false;
```

Comment:
We're cleaning the state before reading here isn't?

Reply:
Yeah I think it should be 2 lines lower, after
```solidity
        bytes32[] memory locators = TransientArrayLib.getBytes32(BATCH_LOCATORS_SLOT);
```